### ISSUE TITLE
SC2329 false positive when calling function via variable

### ISSUE THREAD

**Issue by MichaIng** [koalaman/shellcheck#3273]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2329
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
func1(){ echo 'func1 called'; }
i=1
"func${i}"
exit 0
```

#### Here's what shellcheck currently says:
```console
In test line 2:
func1(){ echo 'func1 called'; }
^-----------------------------^ SC2329 (info): This function is never invoked. Check usage (or ignored if invoked indirectly).
```

#### Here's what I wanted or expected to see:
Nothing, since the function is called. It might be expected behavior, similar to how function calls from traps are not recognized (yet), but at least the wiki does not mention it. There are other checks where `shellcheck` takes into account variables, so probably it is possible here as well for unconditionally assigned variables.

**Comment by litaocdl** [comment]
+1, 

trap function is exposed as not been used.

**Comment by es20490446e** [comment]
I'm experiencing this bug too since today.

Not through `trap` though.

The problematic code is:

```
package_rust-aarch64-gnu () {
	SetAarch64GnuVars
	PackageModule "aarch64-gnu"
}


SetAarch64GnuVars () {
	pkgdesc="Allows using Rust on AArch64 GNU"

	depends=(
		"aarch64-linux-gnu-gcc"
		"aarch64-linux-gnu-glibc"
		"rust"
	)
}
```

`SetAarch64GnuVars` is detected as never being executed.

This is a `PKGBUILD` ([link](https://gitlab.com/es20490446e/express/-/blob/master/recipes/rust/PKGBUILD?ref_type=heads)). Hence `package_rust-aarch64-gnu` is never explicitly executed within the file, but by the tool `makepkg`.

Still the wrong function name is detected.

**Comment by es20490446e** [comment]
This regression was likely introduced in this [commit](https://github.com/koalaman/shellcheck/commit/4f81dbe839091a06a5cfaea695cf1c451ff07565), by @koalaman.

**Comment by hseg** [comment]
This affects traps as well, eg
```sh
trap on_exit INT TERM EXIT
on_exit() { rm -r "${tempfiles[@]}"; }
```

**Comment by MichaIng** [comment]
Right, as mentioned at the end of my post. See also #2542 in these regards.
